maestro-orchestra/src/main/java/maestro/orchestra/yaml/YamlCommandReader.kt

```diff
@@ -38,25 +42,73 @@ class YamlCommandReader : CommandReader {
             }
     }
 
-    override fun readCommands(source: Source): List<MaestroCommand> {
-        val fluentCommands = try {
-            mapper.readValue(
-                source.buffer().inputStream(),
-                object : TypeReference<List<YamlFluentCommand>>() {}
-            )
+    // Files to watch for changes. Includes any referenced files.
+    fun getWatchFiles(flowFile: File): List<File> {
+        return listOfNotNull(
+            flowFile,
+            getInitFlowFile(flowFile)
+        ).filter { it.parentFile.isDirectory }
+    }
+
+    fun readCommands(flowFile: File): Pair<List<MaestroCommand>, List<MaestroCommand>> {
+        return try {
+            val commands = flowFile.source().use {
+                readCommandsUnsafe(it)
+            }
+            val initFlowCommands = getInitCommands(flowFile, commands)
+            initFlowCommands to commands
         } catch (e: MismatchedInputException) {
             val message = e.message ?: throw e
 
             when {
-                message.contains("Unrecognized field") -> throw CommandReader.SyntaxError
-                message.contains("Cannot construct instance") -> throw CommandReader.SyntaxError
-                message.contains("Cannot deserialize") -> throw CommandReader.SyntaxError
-                message.contains("No content to map") -> throw CommandReader.NoInputException
+                message.contains("Unrecognized field") -> throw SyntaxError
+                message.contains("Cannot construct instance") -> throw SyntaxError
+                message.contains("Cannot deserialize") -> throw SyntaxError
+                message.contains("No content to map") -> throw NoInputException
                 else -> throw e
             }
         }
+    }
+
+    private fun readCommandsUnsafe(source: Source) = mapper.readValue(
+        source.buffer().inputStream(),
+        object : TypeReference<List<YamlFluentCommand>>() {}
+    ).map { it.toCommand() }
+
+    private fun getInitCommands(
```

Comment:
Could you make it public? I would benefit from having access to this function on the backend

Reply:
Updated